src/server/pfs/server/api_server.go

```diff
@@ -231,17 +231,24 @@ func (a *apiServer) ListCommit(request *pfs.ListCommitRequest, respServer pfs.AP
 	})
 }
 
+// InspectCommitsetInTransaction performs the same job as InspectCommitset
+// without the option of blocking for commits to finish so that it can run
+// inside an existing postgres transaction.  This is not an RPC.
+func (a *apiServer) InspectCommitsetInTransaction(txnCtx *txncontext.TransactionContext, commitset *pfs.Commitset) ([]*pfs.CommitInfo, error) {
```

Comment:
It is useful to inspect a commitset in a transaction, but we can't block so this provides everything but the blocking functionality of `InspectCommitset`.